src/diffusers/pipelines/marigold/pipeline_marigold_depth.py

```diff
@@ -0,0 +1,1246 @@
+                f"({self.optimal_processing_resolution}). Further memory savings can be achieved with pipeline "
+                f"offloading. Refer to https://huggingface.co/docs/diffusers/optimization/memory for more information."
+            )
+
+        def preset_override(new_denoising_steps: int, new_ensemble_size: int, new_processing_resolution: int):
+            nonlocal denoising_steps, ensemble_size, processing_resolution
+            if denoising_steps is not None:
+                logger.warning(
+                    f"Overriding `denoising_steps`={denoising_steps} due to preset {preset} with "
+                    f"value {new_denoising_steps}."
+                )
+            if ensemble_size is not None:
+                logger.warning(
+                    f"Overriding `ensemble_size`={ensemble_size} due to preset {preset} with value {new_ensemble_size}."
+                )
+            if processing_resolution is not None:
+                logger.warning(
+                    f"Overriding `processing_resolution`={processing_resolution} due to preset {preset} with "
+                    f"value {new_processing_resolution}."
+                )
+            denoising_steps = new_denoising_steps
+            ensemble_size = new_ensemble_size
+            processing_resolution = new_processing_resolution
+
+        def maybe_override(new_denoising_steps: int, new_ensemble_size: int, new_processing_resolution: int):
+            nonlocal denoising_steps, ensemble_size, processing_resolution
+            if denoising_steps is None:
+                denoising_steps = new_denoising_steps
+            if ensemble_size is None:
+                ensemble_size = new_ensemble_size
+            if processing_resolution is None:
+                processing_resolution = new_processing_resolution
+
+        # presets logic
+        if preset == "paper" and generator is None:
+            raise ValueError('`preset` value `"paper"` requires `generator` to be set to ensure reproducibility.')
+        if isinstance(self.scheduler, DDIMScheduler):
+            scheduler = "DDIMScheduler"
+            if preset == "fast":
+                preset_override(10, 1, self.optimal_processing_resolution)
+            elif preset == "precise":
+                preset_override(10, 10, self.optimal_processing_resolution)
+            elif preset == "paper":
+                preset_override(50, 10, self.optimal_processing_resolution)
+            else:
+                # closest to fast
+                maybe_override(10, 1, self.optimal_processing_resolution)
+            assert denoising_steps is not None and denoising_steps > 0
+            if denoising_steps < 10:
+                logger.warning(
+                    f"Detected `denoising_steps`={denoising_steps} with DDIMScheduler; at least 10 is recommended. "
+                    f"Consider using the LCM checkpoint for few-step inference."
+                )
```

Comment:
Oh I see. I think this should be better documented instead of being automatically done by the codebase. Goes a little against our design philosophy: 

> As PyTorch states, explicit is better than implicit and simple is better than complex. This design philosophy is reflected in multiple parts of the library:

Reply:
Without the presets, there is no good default set of kwargs that will ensure smooth operation with both DDIM and LCM checkpoints..

Reply:
Fixed; all preset logic is removed. Default minimums for processing steps are loaded from the model config